t/unit/tasks/test_tasks.py

```diff
@@ -774,6 +778,97 @@ def yyy2():
 
         assert yyy2.__name__
 
+    def test_default_priority(self):
+
+        @self.app.task(shared=False)
+        def yyy3():
+            pass
+
+        @self.app.task(shared=False, priority=66)
+        def yyy4():
+            pass
+
+        @self.app.task(shared=False, bind=True, base=TaskWithPriority)
+        def yyy5(self):
+            pass
+
+        self.app.conf.task_default_priority = 42
+        old_send_task = self.app.send_task
```

Comment:
I don't think you need to save and restore the method object, `self.app` is reinstantiated on each test setup.

Reply:
The restoration part is there because I saw same code in other existing test, and copied it into my test